src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.cs

```diff
@@ -73,6 +73,8 @@ private static bool GetIsWindowsSubsystemForLinux()
             return false;
         }
 
+        public static bool IsIntMaxValueArrayIndexNotSupported => true;
```

Comment:
It would be nice to auto-detect the right value for this, similar to how IsNonZeroLowerBoundArraySupported is auto-detected.

Reply:
I could add that but what's the coreclr way of checking that for this behaviour? Based on my comment above I could catch System.OverflowException but that seems quite odd to me.